### PR TITLE
Document git module return values

### LINKED ISSUE #5379: Document Git module's result
##### ISSUE TYPE

Documentation Report
##### COMPONENT NAME

git
##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

The Git module docs don't indicate that any result is available. In fact, the git module provides "after" and "before" results which are the commit IDs before and after the git command was run (I presume). This should be documented.

### PR DESCRIPTION
**Pull request by bmillemathias**
##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request
##### COMPONENT NAME

source_control/git.py
##### SUMMARY

Document the git module return values `after` and `before`
Fix #5379

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @bmillemathias for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: source_control/git.py [hunk 1]
```diff
@@ -212,6 +212,19 @@
 - git: repo=https://github.com/ansible/ansible-examples.git dest=/src/ansible-examples refspec=+refs/pull/*:refs/heads/*
 '''
 
+RETURN = '''
+after:
+    description: last commit revision of the repository retrived during the update
```
Comment (robinro) [inline source_control/git.py]: Instead of `retrieved` I'd put `checked out`. Fetch without checkout is a rather unusual case.

**Comment by robinro** [comment]
Thanks @bmillemathias for adding this documentation. LGTM overall.

You could also add the non-standard return values:
- `remote_url_changed`:  True/False, True if the remote URL was changed.
- `warnings`: List of warnings if requested features were not available due to a too old git version.

**Comment by bmillemathias** [comment]
Hi @robinro 

the variable `remote_url_changed` is not returned when I test with ansible 2.1x or 2.2. it is always set ? 

Thanks

**Comment by robinro** [comment]
It if only set when the URL is changed. Thats also bad style and should be fixed.

**Comment by rcutmore** [comment]
I submitted a PR [here](https://github.com/ansible/ansible-modules-core/pull/5677) to set `remote_url_changed` whether it's true or false.

**Pull request closed without merging**